framework/src/main/java/org/checkerframework/common/reflection/DefaultReflectionResolver.java

```diff
@@ -546,42 +537,50 @@ private boolean isUnknownMethod(MethodInvocationTree tree) {
         Names names = Names.instance(context);
 
         List<Symbol> result = new ArrayList<>();
-        try {
-            Method loadClass =
-                    Resolve.class.getDeclaredMethod(
-                            "loadClass", Env.class, Name.class, RecoveryLoadClass.class);
-            loadClass.setAccessible(true);
-            Symbol symClass =
-                    (Symbol) loadClass.invoke(resolve, env, names.fromString(className), null);
-            if (!symClass.exists()) {
-                debugReflection("Unable to resolve class: " + className);
-                return Collections.emptyList();
-            }
+        Symbol symClass = getSymbol(className, env, names, resolve);
+        if (!symClass.exists()) {
+            debugReflection("Unable to resolve class: " + className);
+            return Collections.emptyList();
+        }
 
-            ElementFilter.constructorsIn(symClass.getEnclosedElements());
+        ElementFilter.constructorsIn(symClass.getEnclosedElements());
 
-            for (Symbol s : symClass.getEnclosedElements()) {
-                // Check all constructors
-                if (s.getKind() == ElementKind.CONSTRUCTOR) {
-                    // Check for number of parameters
-                    if (((MethodSymbol) s).getParameters().size() == paramLength) {
-                        result.add(s);
-                    }
+        for (Symbol s : symClass.getEnclosedElements()) {
+            // Check all constructors
+            if (s.getKind() == ElementKind.CONSTRUCTOR) {
+                // Check for number of parameters
+                if (((MethodSymbol) s).getParameters().size() == paramLength) {
+                    result.add(s);
                 }
             }
-            if (result.isEmpty()) {
-                debugReflection("Unable to resolve constructor!");
-            }
+        }
+        if (result.isEmpty()) {
+            debugReflection("Unable to resolve constructor!");
+        }
+        return result;
+    }
+
+    private Symbol getSymbol(String className, Env<AttrContext> env, Names names, Resolve resolve) {
+        Method loadClass;
+        try {
+            loadClass =
+                    Resolve.class.getDeclaredMethod(
+                            "loadClass", Env.class, Name.class, RecoveryLoadClass.class);
+            loadClass.setAccessible(true);
+        } catch (SecurityException | NoSuchMethodException | IllegalArgumentException e) {
+            // A problem with javac is serious and must be reported.
+            throw new BugInCF("Error in obtaining reflective method.", e);
+        }
+        try {
+            RecoveryLoadClass noRecovery = (e, n) -> null;
```

Comment:
I had to add this line; otherwise, anonymous classes throw an InvocationTargetException.